File: Week-1/ReverseString-344/solution.js
// Runtime: 112 ms, faster than 71.90% of JavaScript online submissions for Reverse String.
// Memory Usage: 46.9 MB, less than 26.67% of JavaScript online submissions for Reverse String.

var reverseString = function(s) {
  var left = 0,
    right = s.length - 1;
  while (left < right) {
    [s[left], s[right]] = [s[right], s[left]];
    left++;
    right--;
  }
};

File: Week-1/ReverseString-344/solution.cpp
class Solution
{
public:
    void reverseString(vector<char> &s)
    {
        int end = s.size() - 1;
        int temp;
        for (int i = 0; i < s.size() / 2; i++)
        {
            temp = s[i];
            s[i] = s[end];
            s[end] = temp;
            end--;
        }
    }
};